src/shogun/features/DotFeatures.cpp

```diff
@@ -51,9 +51,6 @@ float64_t CDotFeatures::dense_dot_sgvec(int32_t vec_idx1, SGVector<float64_t> ve
 void CDotFeatures::dense_dot_range(float64_t* output, int32_t start, int32_t stop, float64_t* alphas, float64_t* vec, int32_t dim, float64_t b) const
 {
 	ASSERT(output)
-	// write access is internally between output[start..stop] so the following
-	// line is necessary to write to output[0...(stop-start-1)]
-	output-=start;
```

Comment:
output was assumed to be accessed from `start` to `stop`, which wasn't the case.